src/Account.sol

```diff
@@ -28,115 +27,87 @@ contract Account is IAccount, OpsReady, Auth, Initializable {
+    IFactory internal factory;
 
     /*//////////////////////////////////////////////////////////////
                               CONSTRUCTOR
     //////////////////////////////////////////////////////////////*/
 
     /// @notice disable initializers on initial contract deployment
     /// @dev set owner of implementation to zero address
-    /// @param addressResolver: address of Synthetix ReadProxyAddressResolver
-    /// @param marginAsset: address of Synthetix ProxyERC20sUSD
-    /// @param gelato: address of Gelato
-    /// @param ops: address of Ops
+    /// @param _events: address of the contract used by all accounts for emitting events
+    /// @param _marginAsset: address of the Synthetix ProxyERC20sUSD contract used as the margin asset
+    /// @param _futuresMarketManager: address of the Synthetix FuturesMarketManager
+    /// @param _gelato: address of Gelato
+    /// @param _ops: address of Ops
     constructor(
-        address addressResolver,
-        address marginAsset,
-        address gelato,
-        address ops
-    ) Auth(address(0)) OpsReady(gelato, ops) {
+        address _events,
+        address _marginAsset,
+        address _futuresMarketManager,
+        address _systemStatus,
+        address _gelato,
+        address _ops
+    ) Auth(address(0)) OpsReady(_gelato, _ops) {
         // recommended to use this to lock implementation contracts
         // that are designed to be called through proxies
         _disableInitializers();
 
-        ADDRESS_RESOLVER = IAddressResolver(addressResolver);
-        MARGIN_ASSET = IERC20(marginAsset);
+        EVENTS = IEvents(_events);
+        MARGIN_ASSET = IERC20(_marginAsset);
+        FUTURES_MARKET_MANAGER = IFuturesMarketManager(_futuresMarketManager);
+        SYSTEM_STATUS = ISystemStatus(_systemStatus);
     }
 
-    /// @notice initialize contract (only once) and transfer ownership to specified address
-    /// @dev ensure resolver and sUSD addresses are set to their proxies and not implementations
+    /// @notice initialize contract (only once), transfer ownership to specified address,
+    /// and set the address of the Smart Margin Account Factory
     /// @param _owner: account owner
-    /// @param _settings: contract address for account settings
-    /// @param _events: address of events contract for accounts
-    /// @param _factory: contract address for account factory
-    function initialize(
-        address _owner,
-        address _settings,
-        address _events,
-        address _factory
-    ) external initializer {
+    /// @param _factory: address of the Smart Margin Account Factory
+    function initialize(address _owner, address _factory)
+        external
+        initializer
+    {
         owner = _owner;
-        emit OwnershipTransferred(address(0), _owner);
-
-        settings = ISettings(_settings);
-        events = IEvents(_events);
         factory = IFactory(_factory);
```

Comment:
**Suggestion:** If this can be pulled into the constructor for consistency and efficiency that would be awesome, haven't spotted anything else, but will keep looking through. Awesome work! Nice to see the upgradeability issue fixed 😁

Reply:
Fixed 💪